Dip/DipLinuxTests/Sources/main.swift

```diff
@@ -0,0 +1,156 @@
+import Glibc
+import XCTest
+import Dip
+
+
+private protocol Service: class {
+  var client: Client? { get set }
+}
+
+private protocol Client: class {
+  var service: Service { get }
+}
+
+private class ClientImp: Client, Equatable {
+  var service: Service
+  init(service: Service) {
+    self.service = service
+  }
+}
+
+private func ==<T: ClientImp>(lhs: T, rhs: T) -> Bool {
+  return lhs === rhs
+}
+
+private class ServiceImp: Service, Hashable {
+  weak var client: Client?
+  init() {}
+  
+  var hashValue: Int {
+    return unsafeAddressOf(self).hashValue
+  }
+}
+
+private func ==<T: ServiceImp>(lhs: T, rhs: T) -> Bool {
+  return lhs === rhs
+}
+
+
+typealias TMain = @convention(c) (UnsafeMutablePointer<Void>) -> UnsafeMutablePointer<Void>
+
+func dispatch_async(block: TMain) {
+  var pid: pthread_t = 0
+  pthread_create(&pid, nil, block, nil)
+}
+
+func dispatch_sync(block: TMain) -> UnsafeMutablePointer<Void> {
+  var pid: pthread_t = 0
+  var result: UnsafeMutablePointer<Void> = nil
+  pthread_create(&pid, nil, block, nil)
+  pthread_join(pid, &result)
+  return result
+}
```

Comment:
That's only a test code. I did't have a chance to checkout what is there in swift-core-libdispatch, if it is usable I will try to replace it.